ckanext/geocat/utils/ogdch_map_utils.py

```diff
@@ -195,47 +196,82 @@ def get_additonal_relation_protocols():
 
 
 def map_resource(geocat_resource, issued, modified, rights):
+    """
+    map geocat resources to resources on opendata.swiss
+    - issued, modified and rights are taken from the dataset
+    - the title depends on the normed protocol
+    - the format is taken from the geocat resource if it has been set
+    - the media type and the download_url are set only for the normed
+      protocol WWW:DOWNLOAD
+    - url and language are mapped as they come
+    - the original protocol is also stored on the resource since
+      it helps to trace the resource back to the original geocat resource
+    """
     resource_dict = {}
-    title = geocat_resource.get('title', '')
-    resource_dict['title'] = \
-        {'fr': title, 'de': title, 'en': title, 'it': title}
     resource_dict['issued'] = issued
     resource_dict['modified'] = modified
     resource_dict['rights'] = rights
-    resource_dict['format'] = geocat_resource.get('format', '')
+    if geocat_resource.get('format'):
+        resource_dict['format'] = geocat_resource['format']
     resource_dict['description'] = geocat_resource.get('description')
-    resource_dict['media_type'] = geocat_resource.get('media_type', '')
-    name = geocat_resource.get('name')
-    protocol_name = geocat_resource.get('protocol_name')
-    if name and protocol_name and protocol_name.startswith("Map"):
-        resource_dict['title'] = \
-            {'de': protocol_name + " " + geocat_resource['name']['de'],
-             'fr': protocol_name + " " + geocat_resource['name']['fr'],
-             'en': protocol_name + " " + _remove_duplicate_term_in_name(
-                 geocat_resource['name']['en'], "Preview"
-             ),
-             'it': protocol_name + " " + geocat_resource['name']['it'], }
-    elif protocol_name and not name:
-        resource_dict['title'] = \
-            {'de': title,
-             'fr': title,
-             'en': title,
-             'it': title}
-    else:
-        resource_dict['title'] = \
-            {'de': geocat_resource['name']['de'],
-             'fr': geocat_resource['name']['fr'],
-             'en': geocat_resource['name']['en'],
-             'it': geocat_resource['name']['it']}
-    resource_dict['url'] = geocat_resource['url']
-    if geocat_resource['protocol'] == xpath_utils.DOWNLOAD_PROTOCOL:
```

Comment:
This section has been transfered into a function, so it is clearer what goes into this and what is the result.